src/NUnitFramework/framework/Internal/Commands/TimeoutCommand.cs

```diff
@@ -131,27 +152,16 @@ public override TestResult Execute(TestExecutionContext context)
             return context.CurrentResult;
         }
 
-        private Task RunTestWithTimeoutAsync(TestExecutionContext context)
+        private Task<TestResult> RunTestOnSeparateThread(TestExecutionContext context)
         {
-            var testExecution = Task.Run(() => context.CurrentResult = innerCommand.Execute(context));
-
-            AwaitWithoutExceptionWrapping(testExecution, _timeout);
-
-            return testExecution;
+            return Task.Run(() => innerCommand.Execute(context));
         }
 
-        private static void AwaitWithoutExceptionWrapping(Task<TestResult> testExecution, int timeoutMilliseconds)
+        private static void Wait(Task task, int timeoutMilliseconds)
         {
-            var timeoutOrCompletion = Task
-                .WhenAny(testExecution, Task.Delay(timeoutMilliseconds))
-                .Unwrap();
-
-            AwaitCompletionWithoutExceptionWrapping(timeoutOrCompletion);
-        }
+            var timeoutOrCompletion = Task.WhenAny(task, Task.Delay(timeoutMilliseconds));
```

Comment:
This is not new in your change, but why is this this using this Task.WhenAny and that additinal Task.Delay here instead of directly doing `Task.Wait(timeoutMilliseconds)` here?

Reply:
`Task.Wait` throws `AggregateException` if the task in question throws during the wait. In that case, there would be a need for try/catch and `ExceptionDispatchInfo` so that it is possible to rethrow the actual exception thrown by the test.

Reply:
I was not thinking about error handling when looking at that code but now that you spelled it out explicitely it makes sense! 
Thx for the explanation. It is really appreciated.